src/enum_net.cpp

```diff
@@ -650,7 +648,7 @@ namespace libtorrent
 #endif
 
 		SOCKET s = socket(AF_INET, SOCK_DGRAM, 0);
-		if (s == SOCKET_ERROR)
+		if (int(s) == SOCKET_ERROR)
```

Comment:
this is weird.. ``SOCKET`` is a windows type, and ``SOCKET_ERROR`` is a windows value.. it has a different type?